service/history/historyEngine.go

```diff
@@ -696,12 +696,18 @@ Update_History_Loop:
 					failCause = workflow.DecisionTaskFailedCause_BAD_START_TIMER_ATTRIBUTES
 					break Process_Decision_Loop
 				}
+				if !userTimersLoaded {
```

Comment:
how about moving this lazy loading logic within the timerBuilder itself i.e. instead of exposing a LoadUserTimers for lazy loading, just hide it within your timerBuilder and when any of the timer methods are accessed, initiate the lazy load. That way, the code looks clean on the caller.